core/src/main/scala/org/apache/spark/scheduler/BlacklistTracker.scala

```diff
@@ -173,6 +174,24 @@ private[scheduler] class BlacklistTracker (
         listenerBus.post(SparkListenerExecutorBlacklisted(now, exec, newTotal))
         executorIdToFailureList.remove(exec)
         updateNextExpiryTime()
+        // Add executor to blacklist before attempting to kill it. This allows a scheduler backend
+        // to immediately fail to allocate resources on this executor, since killing could be
+        // asynchronous.
+        conf.get(config.BLACKLIST_KILL_ENABLED) match {
+          case Some(enabled) =>
+            if (enabled) {
+              allocationClient match {
+                case Some(allocationClient) =>
+                  logInfo(s"Killing blacklisted executor id $exec" +
+                          s"since spark.blacklist.kill is set.")
```

Comment:
nit: the following line should just be indented (2 spaces), not aligned.  a couple other spots too